src/aliceVision/sfm/pipeline/sequential/ReconstructionEngine_sequentialSfM.cpp

```diff
@@ -437,158 +439,197 @@ void ReconstructionEngine_sequentialSfM::remapLandmarkIdsToTrackIds()
-      }
+            // The beginning of the incremental SfM is a well known risky and
+            // unstable step which has a big impact on the final result.
+            // The Bundle Adjustment is an intensive computing step so we only use it
+            // every N cameras.
+            // We make an exception for the first 'nbFirstUnstableCameras' cameras
+            // and perform a BA for each camera because it makes the results
+            // more stable and it's quite cheap because we have few data.
+            static const std::size_t nbFirstUnstableCameras = 30;
+            int minimalResectionedViewsForBundle = 10;
+            if (_sfmData.getPoses().size() < nbFirstUnstableCameras)
+            {
+                minimalResectionedViewsForBundle = 1;
+            }
 
-      ++resectionId;
-    }
+            // No bundle if we did not accumulate enough resectionned views
+            if(newReconstructedViews.size() < minimalResectionedViewsForBundle && viewsToVisit.size() > 0)
+            {
+                continue;
+            }
 
-    if(_params.rig.useRigConstraint && !_sfmData.getRigs().empty())
-    {
-      ALICEVISION_LOG_INFO("Rig(s) calibration start");
+            triangulate(prevReconstructedViews, newReconstructedViews);
+            bundleAdjustment(newReconstructedViews);
+
+            //Only update prevReconstructedViews after the resectioned views have been refined
+            prevReconstructedViews = _sfmData.getValidViews();
+
+            // Compute the connected views to inform we have new information !
+            registerChanges(linkedViewIds, newReconstructedViews);
+            std::set_union(potentials.begin(), potentials.end(), linkedViewIds.begin(), linkedViewIds.end(), std::inserter(potentials, potentials.end()));
+
+            // scene logging for visual debug
+            // if((resectionId % 3) == 0)
+            // {
+            //     auto chrono_start = std::chrono::steady_clock::now();
+            //     std::ostringstream os;
+            //     os << "sfm_" << std::setw(8) << std::setfill('0') << resectionId;
+            //     sfmDataIO::Save(_sfmData,
+            //                     (fs::path(_sfmStepFolder) / (os.str() + _params.sfmStepFileExtension)).string(),
+            //                     _params.sfmStepFilter);
+            //     ALICEVISION_LOG_DEBUG("Save of file " << os.str() << " took "
+            //                                           << std::chrono::duration_cast<std::chrono::milliseconds>(
+            //                                                  std::chrono::steady_clock::now() - chrono_start)
+            //                                                  .count()
+            //                                           << " msec.");
+            // }
```

Comment:
We could add an option if we don't want to keep it by default.
But we need a way to use it, as it's key for debugging a scene.

It could also make sense to create exports specifically before and after we remove cameras (as it usually means that we are in a bad situation).

Reply:
An option has been added to enable or disable the logging.